src/golang/lib/repos/artifact.go

```diff
@@ -0,0 +1,53 @@
+package repos
+
+import (
+	"context"
+
+	"github.com/aqueducthq/aqueduct/lib/database"
+	"github.com/aqueducthq/aqueduct/lib/models"
+	"github.com/aqueducthq/aqueduct/lib/models/shared"
+	"github.com/google/uuid"
+)
+
+// Artifact defines all of the database operations that can be performed for an Artifact.
+type Artifact interface {
+	artifactReader
+	artifactWriter
+}
+
+type artifactReader interface {
+	// Exists returns whether an Artifact with ID exists.
+	Exists(ctx context.Context, ID uuid.UUID, db database.Database) (bool, error)
+
+	// Get returns the Artifact with ID.
+	Get(ctx context.Context, ID uuid.UUID, db database.Database) (*models.Artifact, error)
+
+	// Get returns the Artifacts with IDs.
+	GetMultiple(ctx context.Context, IDs []uuid.UUID, db database.Database) ([]models.Artifact, error)
+
+	// GetByWorkflowDagId returns the workflows created by workflow dag with ID workflowDagId.
+	GetByWorkflowDagId(ctx context.Context, workflowDagID uuid.UUID, db database.Database) ([]models.Artifact, error)
```

Comment:
nit: Can we rename to `GetByDAG`:
- Just so we can start using `DAG` instead of `WorkflowDag`
- And we can probably drop the `Id` part since its assumed we are using the DAG Id when finding the artifacts for a DAG